pkg/controller/performanceprofile/components/machineconfig/machineconfig.go

```diff
@@ -130,7 +125,11 @@ func getKernelArgs(hugePages *performancev1alpha1.HugePages, isolatedCPUs *perfo
 	}
 
 	if isolatedCPUs != nil {
-		kargs = append(kargs, fmt.Sprintf("isolcpus=%s", string(*isolatedCPUs)))
+		if balanceIsolated != nil && *balanceIsolated == false {
+			kargs = append(kargs, fmt.Sprintf("isolcpus=%s", string(*isolatedCPUs)))
```

Comment:
nit: string(...) is not needed, Sprintf + %s does the trick without explicit cast. But I see it is done at other places as well, so let's do a cleanup after this PR.